src/SelectableBox/index.jsx

```diff
@@ -47,16 +47,17 @@ const SelectableBox = React.forwardRef(({
     ...(type === 'checkbox' && { isIndeterminate }),
   }, null);
 
-  const onClickHandler = () => {
-    inputRef.current.click();
-    if (onClick) { onClick(inputRef.current); }
-  };
+  useEffect(() => {
+    if (onClick && inputRef.current) {
+      inputRef.current.onclick = () => onClick(inputRef.current);
+    }
+  }, [inputRef.current]);
```

Comment:
[clarification] Can you tell me a bit more about why this change was necessary?

Reply:
After upgrading the version of the adapter, there was an inaccuracy in the test coverage of the SelectableBox component. enzyme-adapter-react-17 did track double clicks, while its older version, enzyme-adapter-react-16, did not. Refactoring the SelectableBox component helped us fix the bug we found and make sure the component works correctly.